stl/inc/yvals_core.h

```diff
@@ -19,8 +19,11 @@
 
 #if _STL_COMPILER_PREPROCESSOR
 
+// This does not use `_EMIT_STL_ERROR`, as it needs to be checked before we include anything else.
+// However, `_EMIT_STL_ERROR` has a dependency on `_CRT_STRINGIZE`, defined in `<vcruntime.h>`
 #ifndef __cplusplus
-#error STL1003: Unexpected compiler, expected C++ compiler.
+#pragma message(__FILE__ "(1): STL1003: Unexpected compiler, expected C++ compiler.");

```

Comment:
Should we hard-code this line number to 25 instead of 1? Yes, it will be a PITA to maintain, but I'm concerned that the users who are confused enough to encounter this error will be _really_ confused looking at line 1 and trying to figure out what triggers it.

Reply:
that's a fair enough point; I wish we could use `_CRT_STRINGIZE` here -.-